bgpd/bgp_routemap.c

```diff
@@ -1179,6 +1180,55 @@ static const struct route_map_rule_cmd route_match_vrl_source_vrf_cmd = {
 	route_match_vrl_source_vrf_free
 };
 
+/* `match alias` */
+static enum route_map_cmd_result_t
+route_match_alias(void *rule, const struct prefix *prefix, void *object)
+{
+	char *alias = rule;
+	struct bgp_path_info *path = object;
+	char **communities;
+	int num;
+
+	if (path->attr->community) {
+		frrstr_split(path->attr->community->str, " ", &communities,
+			     &num);
+		for (int i = 0; i < num; i++) {
+			const char *com2alias =
+				bgp_community2alias(communities[i]);
+			if (strncmp(alias, com2alias, strlen(com2alias)) == 0)
```

Comment:
It's not correct to use `strncmp` like this. This condition is true when you have alias `abc` and match by `abcdef`.

Reply:
You must use simple `strcmp` here to make sure that it's exact match.

Reply:
It compares with `strlen` of the _real_ community, not the entered. If I type `match alias some`, but we have `bgp community alias some1`, we compare strlen(some1).

Reply:
I understand, but this doesn't work correctly when we have `match alias some1` and `bgp community alias some` – it returns true in this case.

Reply:
Sounds legit, right.